src/test.cpp

```diff
@@ -43,31 +42,62 @@ int main() { // Test entry point
     return _global_tests.run();
 }
 
+#define STREAM stderr
+#ifdef CI
+#define PRE ""
+#define POST "\n"
+#else
+#define PRE CLEAR "\r"
+#define POST "\r"
+#endif
+
 unsigned int TestSuite::run() {
-    std::printf("Running %d tests\n", num_tests);
+    //TODO(gu) command line flag
+    FILE *report;
+#ifdef REPORT
+    report = std::fopen("report.txt", "w");  // TODO(gu) absolute path
+    if (!report) {
+        std::fprintf(STREAM, "Unable to write report, aborting\n");
+        return 1;
+    }
+#else
+    report = std::fopen("/dev/null", "w");
```

Comment:
This isn't very portable. You could make a macro to wrap if you don't want to write anything. Though.. Macros don't have the best reputation. (Wrap fprintf in a macro, like so: #define print(...) std::fprintf(__VA_ARGS__);) And just remove it if we don't want to log it.